### PR TITLE
Remove internal Optional usage in favor of null checks

### PR DESCRIPTION
**Pull request by watsta**
Fixes gh-7155

I did not modify public APIs, tests, examples and samples. Optional usage in the codebase after this PR:

```
config\src\main\java\org\springframework\security\config\annotation\web\configuration\OAuth2ClientConfiguration.java
config\src\test\java\org\springframework\security\config\annotation\web\configurers\NamespaceHttpAnonymousTests.java
config\src\test\java\org\springframework\security\config\annotation\web\configurers\NamespaceHttpLogoutTests.java
config\src\test\java\org\springframework\security\config\doc\XmlNode.java
config\src\test\java\org\springframework\security\config\web\server\OAuth2ResourceServerSpecTests.java
config\src\test\java\org\springframework\security\config\web\server\ServerHttpSecurityTests.java
oauth2\oauth2-client\src\main\java\org\springframework\security\oauth2\client\web\reactive\function\client\OAuth2AuthorizedClientResolver.java
oauth2\oauth2-client\src\main\java\org\springframework\security\oauth2\client\web\reactive\result\method\annotation\OAuth2AuthorizedClientResolver.java
oauth2\oauth2-client\src\test\java\org\springframework\security\oauth2\client\web\reactive\function\client\ServerOAuth2AuthorizedClientExchangeFilterFunctionTests.java
oauth2\oauth2-client\src\test\java\org\springframework\security\oauth2\client\web\reactive\function\client\ServletOAuth2AuthorizedClientExchangeFilterFunctionTests.java
oauth2\oauth2-core\src\test\java\org\springframework\security\oauth2\core\web\reactive\function\OAuth2BodyExtractorsTests.java
oauth2\oauth2-jose\src\test\java\org\springframework\security\oauth2\jwt\JwtDecodersTests.java
oauth2\oauth2-jose\src\test\java\org\springframework\security\oauth2\jwt\ReactiveJwtDecodersTests.java
oauth2\oauth2-resource-server\src\test\java\org\springframework\security\oauth2\server\resource\introspection\NimbusOpaqueTokenIntrospectorTests.java
oauth2\oauth2-resource-server\src\test\java\org\springframework\security\oauth2\server\resource\introspection\NimbusReactiveOpaqueTokenIntrospectorTests.java
samples\boot\oauth2resourceserver-multitenancy\src\main\java\org\springframework\boot\env\MockWebServerPropertySource.java
samples\boot\oauth2resourceserver-multitenancy\src\main\java\sample\OAuth2ResourceServerSecurityConfiguration.java
samples\boot\oauth2resourceserver-opaque\src\main\java\org\springframework\boot\env\MockWebServerPropertySource.java
samples\javaconfig\hellojs\src\main\java\org\springframework\security\samples\mvc\MessageJsonController.java
```

Note that Optional is still used in
 - `org.springframework.security.oauth2.client.web.reactive.function.client.OAuth2AuthorizedClientResolver`
- `org.springframework.security.oauth2.client.web.reactive.result.method.annotation.OAuth2AuthorizedClientResolver`

As I couldn't think of a nice way to resolve it.

### CODE REVIEW

File: oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/logout/OidcClientInitiatedLogoutSuccessHandler.java [hunk 1]
```diff
@@ -52,25 +51,32 @@ public OidcClientInitiatedLogoutSuccessHandler(ClientRegistrationRepository clie
...
+	private URI endSessionEndpoint(OAuth2AuthenticationToken token) {
 		String registrationId = token.getAuthorizedClientRegistrationId();
-		return Optional.of(
-				this.clientRegistrationRepository.findByRegistrationId(registrationId))
-				.map(ClientRegistration::getProviderDetails)
-				.map(ClientRegistration.ProviderDetails::getConfigurationMetadata)
-				.map(configurationMetadata -> configurationMetadata.get("end_session_endpoint"))
-				.map(Object::toString)
-				.map(URI::create);
+		ClientRegistration clientRegistration = this.clientRegistrationRepository.findByRegistrationId(registrationId);
```
[the remark this answers is not in the document]
Comment (watsta) [reply oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/logout/OidcClientInitiatedLogoutSuccessHandler.java]: Fixed in 6a4425ca

File: samples/boot/oauth2login/src/integration-test/java/org/springframework/security/samples/OAuth2LoginApplicationTests.java [hunk 2]
```diff
@@ -67,7 +67,6 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
```
[the remark this answers is not in the document]
Comment (watsta) [reply samples/boot/oauth2login/src/integration-test/java/org/springframework/security/samples/OAuth2LoginApplicationTests.java]: Reverted in f6f43784

File: config/src/main/java/org/springframework/security/config/annotation/web/configuration/OAuth2ClientConfiguration.java [hunk 3]
```diff
@@ -71,13 +71,17 @@
...
-								.refreshToken()
-								.clientCredentials(configurer ->
-										Optional.ofNullable(this.accessTokenResponseClient).ifPresent(configurer::accessTokenResponseClient))
-								.build();
+								.refreshToken();
+
+				if (this.accessTokenResponseClient != null) {
+					authorizedClientProviderBuilder.clientCredentials(configurer ->
+									configurer.accessTokenResponseClient(this.accessTokenResponseClient));
+				}
```
[the remark this answers is not in the document]
Comment (watsta) [reply config/src/main/java/org/springframework/security/config/annotation/web/configuration/OAuth2ClientConfiguration.java]: Updated in 3fef1670

**Comment by watsta** [comment]
Hi @eleftherias, Thank you for the comments! I updated the PR with the requested changes.

**Comment by eleftherias** [comment]
Thanks for the PR @watsta! This is now merged into master.

**Pull request merged**